pynicotine/downloads.py

```diff
@@ -360,7 +362,7 @@ def _transfer_request_downloads(self, msg):
 
                 transfer = Transfer(username=username, virtual_path=virtual_path, folder_path=folder_path,
                                     status="Queued", size=size, token=token)
-                self.transfers.appendleft(transfer)
+                self.transfers[username + virtual_path] = transfer
```

Comment:
Consider using `token` as key.

Reply:
Tokens are only used for active downloads. We could add a separate active_transfers dict for them eventually, but there were no obvious bottlenecks, and I didn't want to complicate things too much yet.